src/DetailView/DetailView.tsx

```diff
@@ -1,19 +1,139 @@
 import React, { FunctionComponent } from 'react';
-import { FullPokemonAttributes } from '../types/index';
+import { FullPokemonAttributes, Type, Stat, Move } from '../types/index';
+import './DetailView.scss';
+import pokeload from '../assets/pikapokeball.gif';
 
 interface DetailViewProps {
   pokemon: FullPokemonAttributes;
 }
 
-const DetailView: FunctionComponent<DetailViewProps> = ({ pokemon }) => (
-  <div className="view">
-    <h1>
-      #{pokemon.pokemon_id} {pokemon.name}
-    </h1>
-    <img alt={pokemon.name} src={pokemon.sprites.front_default} />
-    <h3>Height: {pokemon.height} decimeters</h3>
-    <h3>Weight: {pokemon.weight} hectograms</h3>
-  </div>
-);
+const typeStyleName = (types: Array<Type>) => {
+  const prefix = 'type-' + (types.length === 1 ? 'single' : 'double') + '-';
+  return prefix + types.map(type => type.type.name).join('-');
+};
+
+const typeHeader = (types: Array<Type>) => {
+  return types.map(type => (
+    <div className={`type-header-${type.type.name} view-header`}>
+      {type.type.name}
+    </div>
+  ));
+};
+
+const heightConverter = (height: number) => {
+  const decimetersPerInch = 3.93701;
+  const totalInches = Math.round(height * decimetersPerInch);
+  const feet = Math.floor(totalInches / 12);
+  const inches = totalInches % 12;
+  let heightString = '';
+  if (feet > 0) {
+    heightString += `${feet}' `;
+  }
+  heightString += `${inches}"`;
+  return heightString;
+};
+
+const weightConverter = (weight: number) => {
+  const poundsPerHectogram = 0.220462;
+  const pounds = Math.round(weight * poundsPerHectogram);
+  return `${pounds} lbs`;
+};
+
+const statFormatter = (stats: Array<Stat>) => {
+  return stats.slice(0, 5).map(stat => (
+    <tr>
+      <td className="align-cell-right">
+        {capitalizeFirstLetter(stat.stat.name)}
+      </td>
+      <td className="align-cell-right">{stat.base_stat}</td>
+    </tr>
+  ));
+};
+
+const moveFormatter = (moves: Array<Move>) => {
```

Comment:
Maybe a comment/doc string for what this is doing? Looks like we're pulling out and displaying 5 random moves?